generate_extracts.py

```diff
@@ -1,55 +1,98 @@
-                path,
-                tile[2],
-                tile[0],
-                tile[1]))
-            if not parent_file.exists():
-                # parent was not build
-                continue
-            if parent_file.stat().st_size < 1.5*10**9:
-                print("Do not build {} because parent is small".format(tile))
-                if parent_file.as_posix() not in tiles:
-                    tiles.append(parent_file.as_posix())
-                continue
-            if tile_path.exists() and parent_file.stat().st_mtime < tile_path.stat().st_mtime:
-                print("Do not build {} because parent is older".format(tile))
-                continue
-            com = "osmconvert -b={},{},{},{} -o={} --complete-ways --complex-ways {}".format(
-                box.west,
-                box.south,
-                box.east,
-                box.north,
-                tile_path.as_posix(),
-                parent_file.as_posix())
-            print(com)
-            #t = Thread(target = process_async, args = (com, ))
-            #threads.append(t)
-            os.system(com)
-        for thread in threads:
-            thread.join()
-print(tiles)
+def parse_args() -> argparse.Namespace:
+    def directory_type(raw: str):
+        p = Path(raw)
+        p.mkdir(exist_ok=True)
+        if not p.is_dir():
+            raise argparse.ArgumentTypeError(f'Path {raw} is not a directory')
+        return p
+
+    parser = argparse.ArgumentParser('generate_extracts',
+                                     description='Extract similarly sized files from the latest OpenStreetMap '
+                                                 'Planet dump.')
+    parser.add_argument('-p', '--planet-dump', dest='planet_dump',
+                        default='https://planet.openstreetmap.org/pbf/planet-latest.osm.pbf',
+                        help='Url of the source pbf file')
+    parser.add_argument('-w', '--working-dir', dest='working_dir', type=directory_type,
+                        default=os.path.join(os.path.dirname(__file__), 'tmp'),
+                        help='Working directory to which the planet dump gets downloaded and in which intermediate '
+                             'split files are stored.')
+    parser.add_argument('-o', '--output-dir', dest='output_dir', type=directory_type,
+                        default=os.path.join(os.path.dirname(__file__), 'out'))
+    parser.add_argument('-s', '--target-size', dest='target_size', default=1.5*10**9, type=int,
+                        help='Target size which the split files should have')
+    return parser.parse_args()
+
+
+def download_planet_dump(url: str, working_dir: Path):
+    print(f'Downloading {url}')
+    subprocess.run(['wget', '-N', '-nv', '--show-progress', url], check=True, cwd=str(working_dir))
+
+    file_name = url.rsplit('/', 1)[-1]
+    subprocess.run(['ln', '-sf', (working_dir / file_name).absolute(), (working_dir / 'source.osm.pbf').absolute()])
```

Comment:
I think you want to name the file 0_0_0.pbf or you have to change the code below accessing the file 0_0_0.pbf

Reply:
i have done so